src/select.rs

```diff
@@ -121,3 +130,18 @@ impl<'a,T> Iterator for FilterNodes<T> where T: Iterator<Item=&'a Node<'a>> {
     }
 }
 
+pub struct TextFilter;
+
+impl<'a,T> Iterator for FilterNodes<T, TextFilter> where T: Iterator<Item=&'a Node<'a>> {
+    type Item = Ref<'a, String>;
+
+    #[inline]
+    fn next(&mut self) -> Option<Ref<'a, String>> {
+        for node in self.iter.by_ref() {
+            if node.is_text() {
+                return Some(node.as_text().unwrap().borrow())
```

Comment:
Please avoid `.unwrap()` and use `if let` instead. (This also makes `is_text` unnecessary.)